### PR TITLE
Utilize WordPress PHPUnit Test Library Included By `wp-env`

### LINKED ISSUE #39863: PHPUnit Container Not Working on M1
### Description

I am currently working on switching our unit testing suite to `wp-env` and have run into a problem. When attempting to run commands in the `phpunit` container I receive a `standard_init_linux.go:228: exec user process caused: exec format error` error.

It appears that the [wordpressdevelop/phpunit](https://hub.docker.com/r/wordpressdevelop/phpunit) image that the `phpunit` container uses does not have an ARM build. During my investigation, I found a [Trac ticket indicating WordPress Core has stopped using the image](https://core.trac.wordpress.org/ticket/54112). I'm planning to work around this issue by running PHPUnit using our Composer dependencies in the CLI container, but given that WordPress Core has stopped using the container, I wanted to highlight the issue I encountered.

### Step-by-step reproduction instructions

1. Have an M1 Mac
2. Install `wp-env`
3. Run `wp-env run phpunit "phpunit"`

### Screenshots, screen recording, code snippet

_No response_

### Environment info

- `wp-env` version: 4.3.1
- Docker is running on a Multipass 1.8.1 container

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
## What?
This makes the appropriate changes to move us away from needing `wp-phpunit/wp-phpunit`. In the process, I also found that `Yoast/PHPUnit-Polyfills` was not being fully utilized. One of the changes was to change methods like `setUp()` to `set_up()` to prevent PHP 8 compatibility issues.

In addition, as per #39863, the `phpunit` container does not work on M1 chips. This pull request replaces the usage of the `phpunit` container with direct usage of `vendor/bin/phpunit` from Gutenberg.

Closes #39863.

## Why?
#41780 made it so that there's a version of WordPress' PHPUnit test files available within the `wp-env` container. We should dogfood this implementation and use the test files here.

## How?
This pull request makes a few changes to methods in order to be compatible with `Yoast/PHPUnit-Polyfills`:
```
* `setUp()` -> `set_up()`
* `tearDown()` -> `tear_down()`
* `setUpBeforeClass()` -> `set_up_before_class()`
* `setUpAfterClass()` -> `set_up_after_class()`
```

This also removes the `wp-phpunit/wp-phpunit` dependency since it's no longer necessary.

We also add the `wptests_` database table prefix as we are running the test suite in the tests-wordpress container rather than the phpunit container. 

## Testing Instructions
1. Make sure CI passes because it's using the same test script.
2. Run `npm run test-unit-php` and verify that it passes.
3. Run `npm run test-unit-php-multisite` and verify that it passes.
4. Play around with different WordPress versions (it looks like we're not compatible as far back as we claim!)

### CODE REVIEW

File: composer.json
Comment (ObliviousHarmony) [inline composer.json:34]: I chose 6.5 here because of the stated requirements for the minimum PHP version. I wanted a version that was compatible with more versions of PHP.

File: package.json
Comment (noahtallen) [inline package.json]: The quotes around verbose -- do those need to be around all of the arguments passed in? Or are they making wp-env pass them into the container?

File: phpunit.xml.dist
Comment (noahtallen) [inline phpunit.xml.dist]: Is there any way to be less heavy-handed? I think this rule could still be useful in general, but I agree we shouldn't warn about the abstract methods.

File: phpunit/multisite.xml
Comment (noahtallen) [inline phpunit/multisite.xml]: This is duplicated from line 12

Comment (ObliviousHarmony) [reply phpunit/multisite.xml]: I really should review all of the files after crazy merges, eh? Thanks!

Comment (noahtallen) [reply phpunit/multisite.xml]: hahaha

File: phpunit.xml.dist
Comment (ObliviousHarmony) [reply phpunit.xml.dist]: I've gone ahead and just marked these tests as incomplete.

File: package.json
Comment (ObliviousHarmony) [reply package.json]: I don't actually know why that was there, I've removed the quotes and it works the same as it did before.

Comment (noahtallen) [reply package.json]: Ah, it must have been left over from when we had to wrap the entire command with quotation marks.

**Review by noahtallen — APPROVED** [review]

**Pull request merged**
